hyperspy/learn/mva.py

```diff
@@ -1317,22 +1369,679 @@ def _estimate_elbow_position(self, curve_values):
         # more or less defines a triangle
         # The elbow should be the point which is the
         # furthest distance from this line
-
-        y2 = np.log(curve_values[maxpoints])
+        if log:
+            y2 = np.log(curve_values[maxpoints])
+            y1 = np.log(curve_values[0])
+        else:
+            y2 = curve_values[maxpoints]
+            y1 = curve_values[0]
         x2 = maxpoints
-        y1 = np.log(curve_values[0])
         x1 = 0
         # loop through the curve values and calculate
         distance = np.zeros(maxpoints)
         for i in range(maxpoints):
-            y0 = np.log(curve_values[i])
+            #y0 = np.log(curve_values[i]+1)
+            if log:
+                y0 = np.log(curve_values[i])
+            else:
+                y0 = curve_values[i]
             x0 = i
             distance[i] = np.abs((x2 - x1) * (y1 - y0) - (x1 - x0) * (y2 - y1)) /\
                 np.math.sqrt((x2 - x1)**2 + (y2 - y1)**2)
         # Point with the largest distance is the "elbow"
         elbow_position = np.argmax(distance)
         return elbow_position
 
+    def scale_data_for_clustering(self,
+                                  use_decomposition_results=True,
+                                  scaling="minmax",
+                                  scaling_kwargs={},
+                                  number_pca_components=None,
+                                  navigation_mask=None,
+                                  signal_mask=None):
+        """scale data for cluster analysis
+
+        Results are stored in `learning_results`.
+
+        Parameters
+        ----------
+        n_clusters : int
+            Number of clusters to find.
+        use_decomposition_results : bool or numpy array
+            If True (recommended) the signal's decomposition results are used
+            for clustering. Note - If this option is not used the raw data
+            is used. This can be memory intensive and is only recommened if
+            the Signal has a small `signal_dimension`.
+        scaling : {"standard","norm","minmax", or scikit learn scaling method}
```

Comment:
I think this should support "None", i.e. no scaling, for the case where the data may already be appropriately scaled, or indeed the data is already in the right range.